core/src/apps/cardano/layout/__init__.py

```diff
@@ -49,13 +49,43 @@ async def confirm_transaction(ctx: wire.Context, amount: int, fee: int, network_
     t1.normal("including fee:")
     t1.bold(format_coin_amount(fee))
 
+    # todo: GK - deposit??
+
     t2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
     t2.normal("Network:")
     t2.bold(network_name)
 
     await require_hold_to_confirm(ctx, Paginated([t1, t2]))
 
 
+async def confirm_certificate(ctx: wire.Context, certificate: CardanoTxCertificateType) -> bool:
+    t1 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    t1.normal("Confirm certificate:")
+    t1.bold(_format_certificate_type(certificate.type))
+    t1.normal("for address:")
+    t1.bold(address_n_to_str(certificate.path))
+
+    if certificate.type == CardanoCertificateType.STAKE_DELEGATION:
+        t2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
```

Comment:
Perhaps I don't understand the UI flow --- but why do we mention "transaction" here when dealing with a single certificate? Also a couple of line above this. If it is just a common headline, we should name the string constant and keep it in only one place.

Reply:
It is just a common headline. I've extracted the whole creation of the text, not just the string. It makes sanse IMHO.